CMakeLists.txt

```diff
@@ -20,7 +22,7 @@ CONFIGURE_FILE("templates/version.hpp.in" "${CMAKE_BINARY_DIR}/generated/include
 
 # Link this 'library' to set the c++ standard / compile-time options requested
 ADD_LIBRARY(project_options INTERFACE)
-TARGET_COMPILE_FEATURES(project_options INTERFACE cxx_std_17)
+TARGET_COMPILE_FEATURES(project_options INTERFACE cxx_std_${CMAKE_CXX_STANDARD})
```

Comment:
is this still needed with the new line 8 above?

Reply:
TARGET_COMPILE_FEATURES approach is I think the better version here because the standard features can then be applied to all libraries and targets. The CMAKE_CXX_STANDARD can also a parameter from command line where this value can be set otherwise it will have a default.